File: handroid/src/main/java/com/h/android/utils/OffRepeatClickUtil.java
package com.h.android.utils;

/**
 * @author zhangxiaohui
 * @describe 防止重复点击
 * @date 2020/5/16
 */
public class OffRepeatClickUtil {
    private static final long MIN_DELAY_TIME = 1000;//最小点击间隔时间
    private static long lastClickTime = 0;

    public static boolean ifCanOnclick(){
        long curTimeMillis = System.currentTimeMillis();
        if(curTimeMillis - lastClickTime > MIN_DELAY_TIME){
            lastClickTime = curTimeMillis;
            return true;
        }
        return false;
    }

    public static boolean ifCannotOnclick(){
        long curTimeMillis = System.currentTimeMillis();
        if(curTimeMillis - lastClickTime > MIN_DELAY_TIME){
            lastClickTime = curTimeMillis;
            return false;
        }
        return true;
    }

}

File: handroid/src/main/java/com/h/android/utils/OffRepeatClickUtil.kt
package com.h.android.utils

/**
 * @author zhangxiaohui
 * @describe 防止重复点击
 * @date 2020/5/16
 */
object OffRepeatClickUtil {
    private const val MIN_DELAY_TIME: Long = 800 //最小点击间隔时间
    private var lastClickTime: Long = 0
    fun ifCannotOnclick(): Boolean {
        val curTimeMillis = System.currentTimeMillis()
        if (curTimeMillis - lastClickTime > MIN_DELAY_TIME) {
            lastClickTime = curTimeMillis
            return false
        }
        return true
    }

    /**
     * 一定时间内不会多次触发
     * @param delay 单位毫秒
     * @return
     */
    fun ifCannotOnclick(delay: Long): Boolean {
        val curTimeMillis = System.currentTimeMillis()
        if (curTimeMillis - lastClickTime > delay) {
            lastClickTime = curTimeMillis
            return false
        }
        return true
    }
}